src/ui.c

```diff
@@ -85,12 +85,20 @@ int_handler()
 }
 
 void
-put_line(int y, char *str, int len, int so)
+put_line(int y, char *str, int len, int so, int mpos, int mlen)
```

Comment:
Ah yes I could do that. Although I'm also not sure underline is the best long term case for this, it was just the easiest to test with during development.